.github/workflows/ghci.yml

```diff
@@ -24,6 +25,11 @@ jobs:
           - os: ubuntu-18.04
             cc: clang
             do_coverity: yes
+          # Build in an Alpine Docker image
+          - os: ubuntu-latest
+            cc: gcc
+            docker_image: alpine:latest
```

Comment:
Any reason the new build uses Ubuntu and Alpine “latest” but all the others use specific versions?

Reply:
The idea of the fixed versions was, to keep testing with older but sill support libs *(assuming breaking changes in our dependencies are rare, but accidentally using new APIs is more likely)*. This is somewhat limited by GHA apparently only ever supporting two versions at a time, dropping support for older ones *(meaning we need to change fixed versions every 2 years, or every 4 years if we skip a version)*. Also, originally there were considerations about the no longer existing cache not dealing well with a sudden OS version change.

For Docker builds, the underlying Ubuntu image doesn't matter, we only want to run Docker with it, so “ubuntu-latest” is fine.
Regarding Alpine, “latest” points to the last stable version and I wasn't sure how long each version is supported, but had the impression it is moving relatively fast and thus used “latest”.
Checking the details: a new stable is released every 6 months and while Alpine's “main” repo of each version is supported for 2 years, but support for the “community” repo ends with the next release *(for comparison: Debian has 3 years of full and additional 2 years of LTS support (+ELTS); Ubuntu LTS overall 6 years)*. I'm not sure what sets those two repos apart, some fairly common things like KDE, GNOME and some GUI programs like eg GIMP are in “community”. But it appears like everything we need for the build is in “main”. It would also be possible to follow the latest stable of the same major version *(since Alpine switched from uClibc to musl in 2014, the major version is 3)*.
If you prefer we could track a specific Alpine release or the major version, but on the other hand there might also be a benefit in having *both* builds in an more conservative, older environment (fixed version Ubuntu,MacOS&Windows build) **and** a build with new-ish dependencies (“latest” Alpine build).

Reply:
Thanks for the explanation. I think that’s fine, and indeed this may even provide a benefit as you mentioned.